durexforth.asm

```diff
@@ -97,13 +97,22 @@ _START = * + 1
 
 ; ----------- macros
 
-!set LINK = 0
-
-!macro BACKLINK {
-    ; it's tempting to add the string as a macro parameter,
-    ; but this does not seem to be supported by ACME.
-    !word	LINK
-    !set	LINK = * - 2
+!set TOP = $6fff
+!set DICTOP = TOP -1
+
+!set BACK = *
+* = DICTOP
+!word 0
+* = BACK
+
+!macro BACKLINK .name , .namesize {

```

Comment:
You can calculate the string length with len() but you still need to include an argument for flags.  I don't think macros can be variadic so I just kept it the same for simplicity.

Reply:
Ok! I think it might be a nice thing to calculate string length automatically, but I agree it is better to keep such a change outside this pull request.